components/eamxx/src/physics/mam/eamxx_mam_wetscav_process_interface.cpp

```diff
@@ -0,0 +1,532 @@
+  // inside a parallel_for.
+  const int nlev = nlev_;
+
+  // Loop over atmosphere columns
+  Kokkos::parallel_for(
+      policy, KOKKOS_LAMBDA(const ThreadTeam &team) {
+        const int icol = team.league_rank();  // column index
+
+        const auto atm = mam_coupling::atmosphere_for_column(dry_atm, icol);
+        // set surface state data
+        // fetch column-specific subviews into aerosol prognostics
+        mam4::Prognostics progs =
+            mam_coupling::aerosols_for_column(dry_aero, icol);
+        // fetch column-specific subviews into aerosol tendencies
+        // Note: we are only updating interstitial aerosols.
+        mam4::Tendencies tends =
+            mam_coupling::interstitial_aerosols_tendencies_for_column(
+                dry_aero_tends, icol);
+        /// shallow_convective_precipitation_production
+        const auto rprdsh_icol = ekat::subview(rprdsh, icol);
+        // deep_convective_precipitation_production
+        const auto rprddp_icol = ekat::subview(rprddp, icol);
+        // deep_convective_precipitation_evaporation
+        const auto evapcdp_icol = ekat::subview(evapcdp, icol);
+        // shallow_convective_precipitation_evaporation =
+        const auto evapcsh_icol = ekat::subview(evapcsh, icol);
+        // deep_convective_cloud_fraction
+        const auto dp_frac_icol = ekat::subview(dp_frac, icol);
+        // shallow_convective_cloud_fraction    =
+        const auto sh_frac_icol = ekat::subview(sh_frac, icol);
+
+        const auto icwmrdp_col  = ekat::subview(icwmrdp, icol);
+        const auto icwmrsh_icol = ekat::subview(icwmrsh, icol);
+        const auto nevapr_icol  = ekat::subview(nevapr, icol);
+        const auto cldt_icol    = ekat::subview(cldt, icol);
+
+        const auto dlf_icol   = ekat::subview(dlf, icol);
+        auto aerdepwetis_icol = ekat::subview(aerdepwetis, icol);
+        auto aerdepwetcw_icol = ekat::subview(aerdepwetcw, icol);
+        auto work_icol        = ekat::subview(work, icol);
+        auto wet_diameter_icol =
+            ekat::subview(wet_geometric_mean_diameter_i, icol);
+        auto dry_diameter_icol =
+            ekat::subview(dry_geometric_mean_diameter_i, icol);
+        auto qaerwat_icol     = ekat::subview(qaerwat, icol);
+        auto wetdens_icol     = ekat::subview(wetdens, icol);
+        const auto prain_icol = ekat::subview(prain, icol);
+
+        mam4::wetdep::aero_model_wetdep(
+            team, atm, progs, tends, dt,
+            // inputs
+            cldt_icol, rprdsh_icol, rprddp_icol, evapcdp_icol, evapcsh_icol,
+            dp_frac_icol, sh_frac_icol, icwmrdp_col, icwmrsh_icol, nevapr_icol,
+            dlf_icol, prain_icol,
+            // outputs
+            wet_diameter_icol, dry_diameter_icol, qaerwat_icol, wetdens_icol,
+            aerdepwetis_icol, aerdepwetcw_icol, work_icol);
+        team.team_barrier();
+        // update interstitial aerosol state
```

Comment:
You might be able to use the field process "combine" here to avoid having to explicitly write out a Kokkos for loop.  See for details https://github.com/E3SM-Project/scream/blob/master/components/eamxx/src/share/util/scream_combine_ops.hpp

Reply:
Thanks, @AaronDonahue. Is it okay if we do it in a follow-up PR? I will add it to the issue I already created.

Reply:
The Field::update methods must be called from outside device code, since they dispatch kernels inside. I think it may be fine to do it inside this pfor, since it's already dispatched...